### PR TITLE
[GLUTEN-7178][VL] Fix field not found error when struct field name contains upper case

### PR DESCRIPTION
**Pull request by zml1206**
## What changes were proposed in this pull request?

Fixes: \#7178

## How was this patch tested?
UT

### CODE REVIEW

File: backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala [hunk 1]
```diff
@@ -326,6 +326,8 @@ object CHBackendSettings extends BackendSettingsApi with Logging {
 
   override def supportStructType(): Boolean = true
 
+  override def structFieldToLowerCase(): Boolean = false
```
Comment (PHILO-HE) [inline backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala:329]: @taiyang-li, curious whether this issue exists in ch backend. If the fix is applicable to both backends, we can remove this separated handling.

Comment (zml1206) [reply backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala:329]: ch backend seems to be different. I didn’t add `structFieldToLowerCase` at first, but ck ci reported an error.

Comment (taiyang-li) [reply backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala:329]: @PHILO-HE  it seems that CH haven't met this kind of issues. Let's keep it to be false for CH backend.

Comment (PHILO-HE) [reply backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala:329]: @taiyang-li, thanks for your verification!

**Review by PHILO-HE — APPROVED** [review]

**Pull request merged**